mypy/fastparse.py

```diff
@@ -56,20 +59,26 @@ def parse(source: Union[str, bytes], fnam: str = None, errors: Errors = None,
         else:
             ast2 = ast27.parse(source, fnam, 'exec')
             ast = conversions.py2to3(ast2)
-    except SyntaxError as e:
-        if errors:
-            errors.set_file('<input>' if fnam is None else fnam)
-            errors.report(e.lineno, e.msg)  # type: ignore
-        else:
-            raise
-    else:
+
         tree = ASTConverter(pyversion=pyversion,
                             is_stub=is_stub_file,
                             custom_typing_module=custom_typing_module,
                             ).visit(ast)
         tree.path = fnam
         tree.is_stub = is_stub_file
         return tree
+    except SyntaxError as e:
+        if errors:
+            errors.set_file('<input>' if fnam is None else fnam)
+            errors.report(e.lineno, e.msg)  # type: ignore
+        else:
+            raise
+    except TypeCommentParseError as e:
```

Comment:
Curious why this can't be written as `except (SyntaxError, TypeCommentParseError) as e:` -- the bodies look identical.